sdks/python/apache_beam/utils/profiler.py

```diff
@@ -59,14 +64,17 @@ def __exit__(self, *args):
 
     if self.profile_location and self.file_copy_fn:
```

Comment:
`self.file_copy_fn` can no longer be None and this check could be removed.

The behavior change here is that now it would not be possible to avoid this code path any more.